src/software/ai/hl/stp/play/tactic_assignment_test.cpp

```diff
@@ -0,0 +1,712 @@
+
+//     friendly_team.assignGoalie(1);
+//     world.updateFriendlyTeamState(friendly_team);
+
+//     auto move_tactic_0 = std::make_shared<MoveTactic>();
+//     auto move_tactic_1 = std::make_shared<MoveTactic>();
+
+//     TacticVector request = {move_tactic_0, move_tactic_1};
+
+//     auto robot_navigation_obstacle_config =
+//     ai_config.robot_navigation_obstacle_config(); GlobalPathPlannerFactory
+//     path_planner_factory(robot_navigation_obstacle_config, world.field()); auto tup =
+//     assignTactics(path_planner_factory, world, request, friendly_team.getAllRobots());
+
+//     auto asst = std::get<2>(tup);
+
+//     EXPECT_EQ(2, asst.size());
+
+//     {
+//         bool has_goalie = false;
+//         bool has_move_0 = false;
+//         for (const auto& [tactic, robot] : asst)
+//         {
+//             if (objectTypeName(*tactic) == TYPENAME(GoalieTactic))
+//             {
+//                 has_goalie = true;
+//                 EXPECT_EQ(robot_1.id(), robot);
+//             }
+//             if (tactic == move_tactic_0)
+//             {
+//                 has_move_0 = true;
+//                 EXPECT_EQ(robot_0.id(), robot);
+//             }
+//         }
+//         EXPECT_TRUE(has_goalie);
+//         EXPECT_TRUE(has_move_0);
+//     }
+
+//     request = {move_tactic_0, move_tactic_1};
+//     tup    = assignTactics(path_planner_factory, world, request,
+//     friendly_team.getAllRobots());
+
+//     asst = std::get<2>(tup);
+
+//     EXPECT_EQ(2, asst.size());
+
+//     {
+//         bool has_goalie = false;
+//         bool has_move_0 = false;
+//         for (const auto& [tactic, robot] : asst)
+//         {
+//             if (objectTypeName(*tactic) == TYPENAME(GoalieTactic))
+//             {
+//                 has_goalie = true;
+//                 EXPECT_EQ(robot_1.id(), robot);
+//             }
+//             if (tactic == move_tactic_0)
+//             {
+//                 has_move_0 = true;
+//                 EXPECT_EQ(robot_0.id(), robot);
+//             }
+//         }
+//         EXPECT_TRUE(has_goalie);
+//         EXPECT_TRUE(has_move_0);
+//     }
+
+//     request = {move_tactic_1, move_tactic_0};
+//     tup = assignTactics(path_planner_factory, world, request,
+//     friendly_team.getAllRobots());
+
+//     asst = std::get<2>(tup);
+//     EXPECT_EQ(2, asst.size());
+
+//     {
+//         bool has_goalie = false;
+//         bool has_move_1 = false;
+//         for (const auto& [tactic, robot] : asst)
+//         {
+//             if (objectTypeName(*tactic) == TYPENAME(GoalieTactic))
+//             {
+//                 //has_goalie = true;
+//                 EXPECT_EQ(robot_1.id(), robot);
+//             }
+//             if (tactic == move_tactic_1)
+//             {
+//                 has_move_1 = true;
+//                 EXPECT_EQ(robot_0.id(), robot);
+//             }
+//         }
+//         EXPECT_TRUE(has_goalie);
+//         EXPECT_TRUE(has_move_1);
+//     }
+// }
```

Comment:
why is this test commented out?

Reply:
These tests do not pass. See https://github.com/UBC-Thunderbots/Software/pull/3091#issuecomment-1865361792